clippy_utils/src/visitors.rs

```diff
@@ -109,30 +109,43 @@ pub fn for_each_expr<'tcx, B, C: Continue>(
         res: Option<B>,
     }
     impl<'tcx, B, C: Continue, F: FnMut(&'tcx Expr<'tcx>) -> ControlFlow<B, C>> Visitor<'tcx> for V<B, F> {
-        fn visit_expr(&mut self, e: &'tcx Expr<'tcx>) {
+        type Result = ControlFlow<()>;
```

Comment:
I've tried this, and the generic rules in the language doesn't permit us to do it because `B` in the internal impl and `B` in the external impl **could** be different (even if it's never actually different) :/

Reply:
I feel like this should be possible, but fine, I can also look into- and try this in a followup, separately, and we can just keep this one as is